fix: resolve Docker file URL networking issue for plugins (#21334)

krikera opened the pull request:
Fixes #21334

### Problem
Using `STORAGE_TYPE=local` and `FILES_URL=http://localhost` no longer works in Docker environments. Plugin daemon cannot access files uploaded by users.

### Solution  
- Added `INTERNAL_FILES_URL` config for Docker network communication
- Plugin file access now uses internal Docker service URLs (e.g., `http://api:5001`)
- User downloads continue using external URLs (e.g., `http://localhost:5001`)
- Simple fallback: uses `FILES_URL` if `INTERNAL_FILES_URL` not set

### Configuration
Add to Docker environment:
```bash
INTERNAL_FILES_URL=http://api:5001
```

### Testing
- ✅ Plugin file access works in Docker
- ✅ User file downloads work from browser  
- ✅ Backward compatible with existing setups

krikera committed:
fix: resolve Docker file URL networking issue for plugins (#21334)

- Add INTERNAL_FILES_URL config for Docker network communication
- Plugin file access now uses internal Docker service URLs
- User downloads continue using external FILES_URL
- Fixes plugin daemon file access in Docker environments
- Backward compatible with existing configurations

Yeuoly reviewed (dismissed):
I guess docker/*compose.yml need to be modified also.

krikera committed:
docs: add INTERNAL_FILES_URL to Docker configuration

- Add INTERNAL_FILES_URL to docker-compose.yaml environment variables
- Add configuration example and documentation to docker/.env.example
- Addresses reviewer feedback for Docker compose file updates

crazywoola committed:
chore: add url to api/

crazywoola committed:
fix: lint
